src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/dto/CalculationResultDto.kt

```diff
@@ -0,0 +1,5 @@
+package uk.gov.justice.digital.hmpps.hmppstier.dto
+
+data class CalculationResultDto constructor(
+  val tierDto: TierDto?
+)
```

Comment:
This is not very nice. The requirement is to check whether the calculation has changed, and the difficulty is when there is no previous calculation. I haven't found an elegant way of doing this and I didn't want to change the TierDto

Reply:
Is this not a case of pulling the existing tier out of the database, calculating the new one, and comaparing
then 
`
old = getOld(crn)
new = calculateNew(crn)
if (old != new) { writebacktodelius() }`

Reply:
yes, as long as the old one is there. I didn't want to expose the DB entity in the listener though. 

Ultimately there's quite a lot going on in the listener, probably need to push some of it down

Reply:
The other complication is supporting the controller, for which we currently have no specs. What the listener needs in order to call the updater does not have a lot in common with what the controller returns